### PR TITLE
chore: Use a token to avoid hitting 60 request rate limit

### PR DESCRIPTION
**Pull request by jsoref**
Signed-off-by: Josh Soref <jsoref@users.noreply.github.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by crenshaw-dev** [comment]
@jsoref when does the "OR" kick in here?

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -311,7 +311,7 @@ jobs:
           file: coverage.out
       - name: Perform static code analysis using SonarCloud
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN || secrets.GITHUB_TOKEN }}
```
Comment (crenshaw-dev) [inline .github/workflows/ci-build.yaml:314]: Probably not this one.  :-)

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yaml (hunk 1 above)
Comment (rishabh625) [reply .github/workflows/ci-build.yaml:314]: @crenshaw-dev : is it possible u to do change and merge this or may be will change post merge

**Comment by jsoref** [comment]
The `||` kicks in if the secret isn't set (or it's empty), which is generally if it's in a forked repository and the fork doesn't have the secret.

File: .github/workflows/ci-build.yaml (hunk 1 above)
Comment (jsoref) [reply .github/workflows/ci-build.yaml:314]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by rishabh625 — APPROVED** [review]
LGTM Thanks!!, just need to fix DCO

**Comment by jsoref** [comment]
I can't really do that before tonight, all I have is my phone.

**Comment by rishabh625** [comment]
i m afraid seeing codecov report

**Comment by crenshaw-dev** [comment]
@rishabh625 the codecov report is so confusing. I think it's analyzing the branch's head commit instead of the hypothetical merge commit? I always go directly to the report and look at the diff there.

**Comment by jsoref** [comment]
I'm used to codecov reports being nonsensical for most changes I make.

**Comment by rishabh625** [comment]
> @rishabh625 the codecov report is so confusing. I think it's analyzing the branch's head commit instead of the hypothetical merge commit? I always go directly to the report and look at the diff there.

Hmm, so merging master into this branch will give proper coverage report?

I was wandering how to get this fixed,as two digit negative % code coverage doesn't looks good

**Comment by crenshaw-dev** [comment]
Yeah I think merging master would make codecov _better_. In my experience, the codecov still isn't 100% correct.

**Comment by blakepettersson** [comment]
What happened with this PR?

**Comment by jsoref** [comment]
Someone else implemented the same changes in #9151

**Pull request closed without merging**